open-craft/opencraft
# [BB-2450] Notice board

This PR adds new page with name `Notice Board` that contains notifications about deployments and implements list endpoint for Open edX deployments.

Mockup itself: https://app.zeplin.io/project/5de4bac9fca2e1782d7245af/screen/5de4c06d0a6f456dd207ed62

**Issue:** [BB-2450](https://tasks.opencraft.com/browse/BB-2450)

**Screenshots**:

_Notification expand and collapse:_

![Peek 2020-07-01 11-19](https://user-images.githubusercontent.com/18251194/86220829-cf8a8a00-bb8c-11ea-96b1-e51356feaa03.gif)

_Different types of notifications:_

![Screenshot_2020-07-01_11-32-54](https://user-images.githubusercontent.com/18251194/86222250-a7038f80-bb8e-11ea-98a1-1ed9b51ee00d.png)


**Testing Instructions:**
1. Checkout this branch.
1. Enter new frontend subdirectory: `cd frontend`
1. Build Ocim API client: `./scripts/build-api-client.sh`
1. Reinstall dependencies: `npm install`
1. Run frontend, `npm start`.
1. Go to `http://localhost:3000/`
1. Log in as user that have instance and at least one deployment.
1. You should see, that `Status & Notifications` navigation link is enabled now:
    ![Screenshot_2020-06-15_12-51-46](https://user-images.githubusercontent.com/18251194/84643802-feef9480-af06-11ea-8cda-e337c412044a.png)
    Click on it, you should see at least one deployment notification.
1. **[optional]**
To test different types of notifications, consider this change:
```diff
diff --git a/registration/api/v2/views.py b/registration/api/v2/views.py
index 2e580b40..527831d9 100644
--- a/registration/api/v2/views.py
+++ b/registration/api/v2/views.py
@@ -211,7 +211,30 @@ class NotificationsViewSet(GenericViewSet):
 
         undeployed_changes = build_instance_config_diff(application)
 
-        notifications = []
+        diff = [
+            ("change", "some_var1", ("test", "test1")),
+            ("change", "theme_config.btn-secondary-bg", ("#000000", "#FFFFFF")),
+            ("add", "theme_config", [("added-bg", "#FFFFFF")]),
+            ("remove", "theme_config", [("removed-bg", "#FFFFFF")]),
+            (
+                "add",
+                "static_content_overrides",
+                [
+                    ("version", 0),
+                    (
+                        "homepage_overlay_html",
+                        "<h1>Welcome to Wonderland</h1><p>It works! Powered by Open edX®</p>",
+                    ),
+                ],
+            ),
+            ("change", "some_dict.version", (0, 1)),
+            ("add", "", [("instance_name", "Wonderland")]),
+            ("remove", "", [("some_var2", "test")]),
+        ]
+        notifications = [
+            {"deployed_changes": diff, "date": application.instance.created, "status": status}
+            for status in DeploymentState
+        ]
         for deployment in deployments:
             deployment_status = deployment.status()
             notifications.append({
```
[Gist with code of endpoint with this change](https://gist.github.com/0x29a/4295e9864110723a59aa830ad612eb03).

**Reviewers**
